### PR TITLE
refactor: remove cdk compat-layer, introduce migrations

### PR DESCRIPTION
**Pull request by hoebbelsB**
# Description

* [x] move `template/schematics/common` to `tools/generators/utils`
* [x] implement cdk migration schematic
* [x] implement tests for cdk migration schematic

### CODE REVIEW

File: libs/cdk/schematics/migration.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "$schema": "../../../node_modules/@angular-devkit/schematics/collection-schema.json",
+  "schematics": {
+    "update-1-0-0": {
```
Comment (edbzn) [inline libs/cdk/schematics/migration.json:4]: I think we should name it `1.0.0-alpha.12`.

File: libs/cdk/schematics/src/migrations/update-1-0-0/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+import * as path from 'path';
+
+describe('cdk migration 1.0.0', () => {
+  let appTree: UnitTestTree;
+
+  /* Increase Jest timeout because of long living tests. */
+  beforeAll(() => jest.setTimeout(10_000));
+
+  /* Reset default timeout to 5s. */
+  afterAll(() => jest.setTimeout(5_000));
```
Comment (edbzn) [inline libs/cdk/schematics/src/migrations/update-1-0-0/index.spec.ts:15]: :confused: I don't know if it's worth investing time to profile the migration to remove this. What do you think?

File: libs/template/schematics/src/commands/ng-add/index.spec.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import {
 } from '@angular-devkit/schematics/testing';
 import { join } from 'path';
 
-import { readJsonInTree } from '../../common';
+import { readJsonInTree } from '../../../../../../tools/generators/common';
```
Comment (edbzn) [inline libs/template/schematics/src/commands/ng-add/index.spec.ts:7]: We can not use path mapping here?

File: libs/cdk/schematics/src/migrations/update-1-0-0/index.spec.ts (hunk 2 above)
Comment (hoebbelsB) [reply libs/cdk/schematics/src/migrations/update-1-0-0/index.spec.ts]: what do you mean? to improve the tests? i won't :D

File: libs/template/schematics/src/commands/ng-add/index.spec.ts (hunk 3 above)
Comment (hoebbelsB) [reply libs/template/schematics/src/commands/ng-add/index.spec.ts:7]: no

**Comment by hoebbelsB** [comment]
@edbzn i've noticed there are new test cases which are failing now. Did u already started working on how to solve them? For now we can only fix imports, but no usages? Or do I miss something?

File: libs/cdk/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@rx-angular/cdk",
-  "version": "1.0.0-alpha.11",
+  "version": "1.0.0-alpha.12",
```
Comment (edbzn) [inline libs/cdk/package.json:3]: @hoebbelsB we should not update the lib manually on a feature branch, I revert this.

File: .github/workflows/build-and-test.yml [hunk 5]
```diff
@@ -104,7 +104,8 @@ jobs:
         run: yarn nx run ssr-e2e:e2e --headless
       # don't run build in parallel, can cause dead locks
       - name: Run Affected build
-        run: yarn nx affected:build
+      # @todo: @edbzn Nx cloud fail, disable it temporary.
+        run: NX_CLOUD_DISTRIBUTED_EXECUTION=false yarn nx affected:build
```
Comment (hoebbelsB) [inline .github/workflows/build-and-test.yml:108]: why? :(

Comment (hoebbelsB) [reply .github/workflows/build-and-test.yml:108]: and if we change the github actions, can we maybe disable `NX_VERBOSE_LOGGING` as well?

File: apps/demos/src/app/features/experiments/structural-directives/if-visible-poc/if-visible.component.ts [hunk 6]
```diff
@@ -1,5 +1,4 @@
-import { ChangeDetectionStrategy, Component } from '@angular/core';
-import { RxStrategyProvider } from '@rx-angular/cdk';
+import { ChangeDetectionStrategy, Component } from '@angular/core';;
```
Comment (hoebbelsB) [inline apps/demos/src/app/features/experiments/structural-directives/if-visible-poc/if-visible.component.ts:1]:
```suggestion
import { ChangeDetectionStrategy, Component } from '@angular/core';
```

File: libs/cdk/project.json [hunk 7]
```diff
@@ -25,6 +25,45 @@
...
+      "dependsOn": [
+        {
+          "target": "build-base",
+          "projects": "self"
+        },
+        {
+          "target": "build-schematics",
+          "projects": "self"
+        }
+      ],
```
Comment (hoebbelsB) [inline libs/cdk/project.json:38]: we could configure this as general taskDependency in `nx.json` what do u think?

File: libs/template/project.json [hunk 8]
```diff
@@ -36,16 +42,18 @@
...
+      "dependsOn": [
+        {
+          "target": "build-lib",
+          "projects": "self"
+        },
+        {
+          "target": "build-schematics",
+          "projects": "self"
+        }
+      ],
```
Comment (hoebbelsB) [inline libs/template/project.json:54]: we could configure this as general taskDependency in nx.json what do u think?

**Review by hoebbelsB — COMMENTED** [review]
@edbzn big THX for jumping into this! tried in another project, works great!

File: libs/cdk/project.json (hunk 7 above)
Comment (edbzn) [reply libs/cdk/project.json:38]: There's an issue with Nx, if I change `cdk:build-base`  to `template:build-lib`  which is the same as the `template:build-lib` then it breaks the build, we cannot use global dependsOn config for now.

**Pull request merged**